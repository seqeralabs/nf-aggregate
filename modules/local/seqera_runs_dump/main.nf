include { getRunMetadata } from './functions'

process SEQERA_RUNS_DUMP {
    tag "$meta.id"
    conda 'tower-cli=0.9.2'
    container 'seqeralabs/nf-aggregate:tower-cli-0.9.2--hdfd78af_1'

    input:
    val meta
    val api_endpoint
    val java_truststore_path
    val java_truststore_password

    output:
    tuple val(metaOut), path("${prefix}"), emit: run_dump
    path "versions.yml"                  , emit: versions

    script:
    def args = task.ext.args ?: ''
    def args2 = task.ext.args2 ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"
    metaOut = meta + getRunMetadata(meta, log, api_endpoint, java_truststore_path, java_truststore_password)
    def fusion = metaOut.fusion ? '--add-fusion-logs' : ''
    javaTrustStore = java_truststore_path ? "-Djavax.net.ssl.trustStore=${java_truststore_path}" : ''
    javaTrustStorePassword = java_truststore_password ? "-Djavax.net.ssl.trustStorePassword=${java_truststore_password}" : ''
    """
    tw \\
        $args \\
        $javaTrustStore \\
        $javaTrustStorePassword \\
        --url=${api_endpoint} \\
        --access-token=$TOWER_ACCESS_TOKEN \\
        runs \\
        dump \\
        -id=${meta.id} \\
        --workspace=${meta.workspace} \\
        --output="${prefix}.tar.gz" \\
        $fusion \\
        $args2

    mkdir -p ${prefix}
    tar \\
        -xvf \\
        ${prefix}.tar.gz \\
        -C ${prefix}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        seqera-cli: \$(echo \$(NO_COLOR=true tw --version 2>&1) | sed 's/^.*Tower CLI version //; s/ *\$//')
    END_VERSIONS
    """
}
