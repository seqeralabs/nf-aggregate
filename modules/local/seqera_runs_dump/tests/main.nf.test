nextflow_process {

    name "Test Process SEQERA_RUNS_DUMP"
    script "../main.nf"
    process "SEQERA_RUNS_DUMP"

    tag "modules"
    tag "modules_local"
    tag "seqera_runs_dump"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = ['id': '4Bi5xBK6E2Nbhj', 'workspace': 'community/showcase']
                input[1] = "https://api.tower.nf"
                """
            }
        }

        then {
            def dumpFiles = file(process.out.run_dump.get(0).get(1)).listFiles().sort()

            assertAll(
                { assert snapshot(
                    dumpFiles.grep { file -> ! file.name.matches("workflow-launch.json") },
                    process.out.workflow_json
                ).match()},
                { assert dumpFiles.find { it.name.matches('workflow-launch.json')}.exists()},
                { assert process.success }
            )
        }

    }

}
