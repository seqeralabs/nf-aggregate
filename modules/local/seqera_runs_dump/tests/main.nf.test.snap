{
    "Should run without failures": {
        "content": [
            [
                "service-info.json:md5,6988ae3ef3c822007075726ecc73ba9c",
                "workflow-load.json:md5,4f02d5a24ab89aa648cd4346785c8f2c",
                "workflow-metadata.json:md5,b37b4faeddf283a2c44cbe4000e4ab6e",
                "workflow-metrics.json:md5,13a5b5d7447fad4a8baa053d1abf85e5",
                "workflow-tasks.json:md5,577a7472816b7729012a9291d97ff150",
                "workflow.json:md5,658701f8d7af1b7112534a7f58cd9aa7"
            ],
            null
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-25T10:21:06.447483"
    }
}